src/python/pants/backend/python/target_types_rules.py

```diff
@@ -143,29 +153,160 @@ async def inject_pex_binary_entry_point_dependency(
 # -----------------------------------------------------------------------------------------------
 
 
+def _classify_entry_points(
+    entry_points: FrozenDict[str, FrozenDict[str, str]]
+) -> Generator[Tuple[bool, str, str, str], None, None]:
+    """Looks at each entry point to see if it is a target address or not.
+
+    Returns a tuple of tuples: IsTarget, section, name, ref.
+    """
+    for section, values in entry_points.items():
+        for name, ref in values.items():
+            yield ref.startswith(":") or "/" in ref, section, name, ref
+
+
+@rule(desc="Determining the entry points for a `python_distribution` target")
+async def resolve_python_distribution_entry_points(
+    request: ResolvePythonDistributionEntryPointsRequest,
+) -> ResolvedPythonDistributionEntryPoints:
+    field_value = request.entry_points_field.value
+    if field_value is None:
+        return ResolvedPythonDistributionEntryPoints()
+
+    address = request.entry_points_field.address
+    classified_entry_points = list(_classify_entry_points(field_value))
+
+    # Pick out all target addresses up front, so we can use MultiGet later.
+    #
+    # This calls for a bit of trickery however (using the "y_by_x" mapping dicts), so we keep track
+    # of which address belongs to which entry point.
+    target_refs = [ref for is_target, _, _, ref in classified_entry_points if is_target]
+    # intermediate step, as Get(Targets) returns a deduplicated set..
+    target_addresses = await Get(
+        Addresses, UnparsedAddressInputs(target_refs, owning_address=address)
+    )
+    address_by_ref = dict(zip(target_refs, target_addresses))
+    targets = await Get(Targets, Addresses, target_addresses)
+
+    # Check that we only have targets with a pex entry_point field.
+    for target in targets:
+        if not target.has_field(PexEntryPointField):
+            raise InvalidFieldException(
+                f'The "entry_points" target address {target.address} does not resolve '
+                f"to a pex_binary, but: {target.alias}"
+            )
+
+    binary_entry_points = await MultiGet(
+        Get(
+            ResolvedPexEntryPoint,
+            ResolvePexEntryPointRequest(target[PexEntryPointField]),
+        )
+        for target in targets
+    )
+    binary_entry_point_by_address = {
+        target.address: entry_point for target, entry_point in zip(targets, binary_entry_points)
+    }
+
+    entry_points: dict[str, dict[str, EntryPoint]] = defaultdict(dict)
+
+    # parse refs/replace with resolved pex entry point, and validate console entry points have function
```

Comment:
Style nit: comments should be sentences.

```suggestion
    # Parse refs/replace with resolved pex entry point, and validate console entry points have function.
```